### DISCUSSION TITLE
Mapgen documentation updating

### DISCUSSION THREAD

**Discussion by NetSysFire** [CleverRaven/Cataclysm-DDA#66549] [General]
As just discussed in the devcord, our docs, especially since we have https://docs.cataclysmdda.org now, are in dire need of updating.
This discussion should be used to coordinate what needs to be updated.

- [ ] The available guides at https://github.com/CleverRaven/Cataclysm-DDA/tree/master/doc/JSON_Mapping_Guides need to be merged, especially since the gutters are unified now
- [ ] place_monster/place_monsters/monsters (and similar for all other directives) is super inconsistently used and poorly described
- [ ] Meta: Many things are duplicated
- [ ] It appears all the `boost*` directives in `regional_map_settings.json` are useless now and it is an ancient relic

Please comment more below and I shall update this list accordingly. Also all folks who have the permissions to edit this discussion, feel free to do so to add stuff.

**Comment by Procyonae** [comment]
List of all the mapping related docs I can think of:
https://docs.cataclysmdda.org/MAPGEN.html
https://docs.cataclysmdda.org/OVERMAP.html
https://docs.cataclysmdda.org/JSON_Mapping_Guides/Guide_for_beginning_mapgen.html
https://docs.cataclysmdda.org/JSON_Mapping_Guides/Guide_for_intermediate_mapgen.html
https://docs.cataclysmdda.org/JSON_Mapping_Guides/JSON_ROOF_MAPGEN.html
https://docs.cataclysmdda.org/TER_FURN_TRANSFORM.html

Related docs that should be linked in more places:
https://docs.cataclysmdda.org/ITEM_SPAWN.html
https://docs.cataclysmdda.org/JSON_INFO.html

Just generally a lot of stuff that could be links isn't

As a side note it would probably be useful if the docs filenames matched up with their markdown titles, and alot of the wiki repeats stuff too and could probably be replaced by links to docs or added to existing docs eg https://github.com/CleverRaven/Cataclysm-DDA/wiki/New-Contributor-Guide-Locations
I like your idea of a doc with testing practises too, stuff like debug -> (m)ap... -> (M)ap editor -> (o)vermap and then using regenerate repeatedly to see different nests etc is really useful and undocumented afaik for example

**Comment by Procyonae** [comment]
I found an undocumented add field in \data\json\mapgen\cs_internet_cafe.json `"add": [ { "item": "mop", "x": [ 11, 12 ], "y": [ 17, 20 ], "chance": 2 } ]`

**Comment by RenechCDDA** [comment]
mapgen_function_json_base::setup_common contains most of the json reads, which can be cross-referenced against the docs for missing fields.

Here's some undocumented ones (read: not documented in mapgen.md):

place_remove_all

add (apparently *depreciated*)

place_signs

place_vendingmachines

place_toilets (mentioned in mapgen.md but doesn't list what it does or how to use it)

place_gaspumps (ditto)

place_items (needs explicit documentation instead of inference)

place_loot (mentioned but no idea what differentiates it from other place_ entries)

remove_npcs

place_traps

place_furniture (?!)

place_terrain (why... why would this be a thing?)

place_graffiti

place_zones

place_ter_furn_transforms


Palettes similarly have undocumented entries in mapgen_palette::load_internal
